src/global/signal_handler.cc

```diff
@@ -308,9 +356,9 @@ void SignalHandler::register_handler(int signum, signal_handler_t handler, bool
   struct sigaction act;
   memset(&act, 0, sizeof(act));
 
-  act.sa_handler = handler_hook;
+  act.sa_handler = (signal_handler_t)handler_signal_hook;
   sigfillset(&act.sa_mask);  // mask all signals in the handler
-  act.sa_flags = oneshot ? SA_RESETHAND : 0;
```

Comment:
Oh, we probably want to preserve the SA_RESETHAND if oneshot?  IIRC this is for SIGSEGV?